docs/extensions/containerruntime.md

```diff
@@ -0,0 +1,37 @@
+# Gardener Container Runtime Extension
+
+At the lowest layers of a Kubernetes node is the software that, among other things, starts and stops containers. It is called “Container Runtime”. 
+The most widely known container runtime is Docker, but it is not alone in this space. In fact, the container runtime space has been rapidly evolving. 
+
+Kubernetes support different container runtimes using  Container Runtime Interface (CRI) – a plugin interface which enables kubelet to use a wide variety of container runtimes.
+
+Gardener supports creation of Worker machines using CRI, more information can be found here: [CRI Support](https://https://github.com/gardener/gardener/blob/master/docs/extensions/operatingsystemconfig.md)).
+
+## Motivation
+
+Prior to the `Container Runtime Extensibility` concept, Gardener used Docker as the only
+container runtime to use in Worker machines. Because of the wide variety of different container runtimes
+offers multiple important features (For example enhanced security concepts) it is important to enable end users to use other container runtimes as well.
+
+## The ContainerRuntime Extensions Resource
+
+Here is what a typical `ContainerRuntime` resource would look-like:
+
+```yaml
+---
+apiVersion: extensions.gardener.cloud/v1alpha1
+kind: ContainerRuntime
+metadata:
+  name: my-container-runtime
+spec:
+  binaryPath: /var/bin/containerruntimes
+  type: gvisor
+```
+
+## Supporting a new Container Runtime Provider
+
+To add support for another container runtime (e.g., gvisor, kata-containers, etc.) a container runtime extension controller needs to be implemented. It should support Gardener's supported CRI plugins.
+
+The container runtime extension should install the necessary resources in the shoot and copy the runtime binaries to the relevant Worker machines in path: `spec.binaryPath`. 
```

Comment:
```suggestion
The container runtime extension should install the necessary resources into the shoot cluster (e.g., `RuntimeClass`es), and it should copy the runtime binaries to the relevant worker machines in path: `spec.binaryPath`. 
```

Reply:
Can you elaborate here on the labels that Gardener adds to every "relevant worker machine" so that extension developers know how to find them? Also, I guess makes sense to shortly explain the example of how to install such binaries on a node (run a DaemonSet that copies them over from a Docker registry, avoid downloading from internet to also cater with isolated environments).